### PR TITLE
Describe the Node ID format

### PR DESCRIPTION
**Pull request by no2chem**
## Overview

Description: This documentation only PR describes the format of Node IDs in Corfu.

Why should this be merged: This simple PR describes only the Node ID format and does not contain code. It is required to safely enable all the clustering features in Corfu.

Related issue(s) (if applicable): Partially addresses #1051

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **57ca671**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #1069 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/6lomr3MK8kjPYc2iNvmz3r8WJyfbjUMx)

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **6032196**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #1069 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/awlF4jFDsMRqQyMrgxeRxsdnUej0Ix9R)

File: docs/NODE_FORMAT.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+This document describes the format of Corfu node identifiers (node IDs), which are used to refer to unique Corfu instances.
+Node Ids provide an extra layer of reliability in the case of a server or network reconfiguration. For example, a DNS error
+could point a Corfu client at an incorrect node. This could lead a client to read an update from the wrong system, or believe
+that it has committed an update when it is not committed. Node IDs prevent this issue by assigning each node a unique 128-bit
+identifier at startup. When clients connect to a Corfu node, they verify that the correct Node ID is presented by the node
+before interacting with it.
+
+Node IDs consist of the following components:
+
+- ``<protocol>``: The protocol that used by the node. Currently, only ``tcp`` is supported.
```
Comment (annym) [inline docs/NODE_FORMAT.md]: that is used... or: The protocol used by...

File: docs/NODE_FORMAT.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+## Corfu Node IDs
+
+This document describes the format of Corfu node identifiers (node IDs), which are used to refer to unique Corfu instances.
+Node Ids provide an extra layer of reliability in the case of a server or network reconfiguration. For example, a DNS error
+could point a Corfu client at an incorrect node. This could lead a client to read an update from the wrong system, or believe
+that it has committed an update when it is not committed. Node IDs prevent this issue by assigning each node a unique 128-bit
+identifier at startup. When clients connect to a Corfu node, they verify that the correct Node ID is presented by the node
+before interacting with it.
+
+Node IDs consist of the following components:
```
Comment (annym) [inline docs/NODE_FORMAT.md]: I think it is a bit confusing in the way its written. If the node ID provides a way of uniqueness and it is "composed" of: the protocol + host name (being this an IP address) + ... and so on... With every change of IP, the node ID changes (though being the same instance)... Isn't the Node ID only the "id" component of this list you are providing? Maybe it is just a naming thing, we are using 'node Id' for both the unique part: "id" and also for the complete 'identifier': protocol://host:port/id

Comment (no2chem) [reply docs/NODE_FORMAT.md]: Hm, yeah let me fix the docs. I renamed things to - NodeLocator when I wrote the actual code.

File: docs/NODE_FORMAT.md (hunk 1 above)
Comment (no2chem) [reply docs/NODE_FORMAT.md]: thanks@

**Pull request merged**